week-3/slideshow/slideshow.js

```diff
@@ -1,8 +1,72 @@
 const images = [
-    "./assets/cute-cat-a.png",
-    "./assets/cute-cat-b.jpg",
-    "./assets/cute-cat-c.jpg",
+  "./assets/cute-cat-a.png",
+  "./assets/cute-cat-b.jpg",
+  "./assets/cute-cat-c.jpg",
+  "./assets/cute-cat-d.jpg",
 ];
 
+// Write your code here
 
-// Write your code here
\ No newline at end of file
+let index = 0;
+const interval = 1000;
+const carouselImage = document.getElementById("carousel-img");
+const imageIndex = document.getElementById("img-index");
+
+const forwardBtn = document.getElementById("forward-btn");
+forwardBtn.addEventListener("click", function () {
+  nextImage();
+});
+
+const backwardsBtn = document.getElementById("backward-btn");
+backwardsBtn.addEventListener("click", function () {
+  previousImage();
+});
+
+function setCarouselImage() {
+  carouselImage.src = images[index];
+  imageIndex.innerText = index;
+}
+
+function nextImage() {
+  index++;
+  if (index == images.length) {
+    index = 0;
+  }
+  setCarouselImage();
+}
+
+function previousImage() {
+  index--;
+  if (index == -1) {
+    index = images.length - 1;
+  }
+  setCarouselImage();
+}
+
+let forwardIntervalId;
+const autoForwardBtn = document.getElementById("auto-forward-btn");
+autoForwardBtn.addEventListener("click", function () {
+  stopIntervals();
+  forwardIntervalId = setInterval(() => {
+    nextImage();
+  }, interval);
+});
+
+const stopBtn = document.getElementById("stop-btn");
+stopBtn.addEventListener("click", function () {
+  stopIntervals();
+});
+
+function stopIntervals() {
+  clearInterval(forwardIntervalId);
+  clearInterval(backwardIntervalId);
+}
+
+let backwardIntervalId;
+const autoBackwardBtn = document.getElementById("auto-backward-btn");
+autoBackwardBtn.addEventListener("click", function () {
+  stopIntervals();
+  backwardIntervalId = setInterval(() => {
+    previousImage();
+  }, interval);
+});
```

Comment:
This code looks like its doing it's job properly, is broken into functions and is easy understand :) I think you're really getting this!

An improvement for next time would be to refactor it a bit. It has loads of very similar code like the back and forwards functions which we could to use one function for both so that the code isn't  duplicated. Not sure if you've seen it in the curriculum but there's a programming principle called DRY Work looking into